src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -5,3 +5,265 @@ title: Build System
+
+However, some modules generate their configuration file at build-time, making it impossible to edit the file in the source tree. In this situation, you can use `./b edit` to open an interactive window with a text editor allowing you to edit the configuration file in the build tree.
+
+One module that generates it configuration file at build time is `support::loggingg::DataLogging`, to edit the configuration file for `DataLogging` run
+
+```sh
+./b edit config/DataLogging.yaml
+```
+
+The `edit` command uses the editor that is defined in your host shell (using the `EDITOR` environment variable). If `EDITOR` is not set then it will default to `nano`. Currently, the only supported editors are `vim` and `nano`, with `nano` used as the fallback in all cases.
+
+##### Format
+
+NUbots uses three different code formatters. `clang-format` is used to format C++ and protobuf files, `black` and `isort` are used to format python files, and `cmake-format` is used to format CMake files.
+
+If you want to ensure that all files in the codebase are formatted according to our defined style run
+
+```sh
+./b format
+```
+
+##### Install
+
+Once code is built and you are ready to install it on a robot, you can run
+
+```sh
+./b install [options] <robot>
+```
+
+`<robot>` can either be a known robot designation (`n1` or `nugus1`, for instance) or an IP address of a robot.
+
+`[options]` may be one, or more, of the following
+
+| Option | Description                                                                    |
+| :----: | :----------------------------------------------------------------------------- |
+|   -u   | The user to install to on the target. Defaults to the user in the Docker image |
+|   -t   | Install toolchain to the target                                                |
+|  -cn   | Only install new config files. This is the default                             |
+|  -cu   | Update config files on the target that are older than the local files          |
+|  -co   | Overwrite all config files on the target                                       |
+|  -ci   | Ignore all changes to config files (installs no config files)                  |
+
+##### Module
+
+The `module` command allows you to generate a new module. To generate a new line detection module one would run
+
+```sh
+./b module generate vision/LineDetector
+```
+
+This will generate all necessary module files in the `module/vision/LineDetector` folder. This includes an empty configuration file, a basic C++ source and header file implementing the `LineDetector` reactor, and an empty unit test file.
+
+##### Run
+
+To run roles locally on your system use the `run` command of `b` as follows
+
+```sh
+./b run <name of role>
+```
+
+---
+
+**NOTE**
+Be sure that you are using the `generic` target for this (`./b target generic`), otherwise you role will likely crash.
```

Comment:
Unless your computer is an Intel 7th gen or newer (at least as new as the robot)